lib/iris/fileformats/pp_rules.py

```diff
@@ -1004,18 +1014,54 @@ def _all_other_rules(f):
                 f.x,
                 standard_name='time',
                 units=cf_units.Unit('days since 0000-01-01 00:00:00',
-                                    calendar=cf_units.CALENDAR_360_DAY),
+                           calendar=cf_units.CALENDAR_360_DAY),
                 bounds=f.x_bounds),
              1))
 
+    if (len(f.lbcode) == 5 and f.lbcode[-1] == 3 and f.lbcode.iy == 23):
+        epoch_days_unit = cf_units.Unit('days since 0000-01-01 00:00:00', 
+                calendar=cf_units.CALENDAR_360_DAY)
+        t1_epoch_days = epoch_days_unit.date2num(f.t1)
+        t2_epoch_days = epoch_days_unit.date2num(f.t2)
+        # The end time is exclusive, not inclusive.
+        dim_coords_and_dims.append(
+            (DimCoord(
+                np.linspace(t1_epoch_days, t2_epoch_days, f.lbrow, endpoint=False),
+                standard_name='time',
+                units=epoch_days_unit,
+                bounds=f.y_bounds),
+             0))
+
     # Site number (--> scalar coordinate)
-    if (len(f.lbcode) == 5 and f.lbcode[-1] == 1 and f.lbcode.ix == 13 and
-            f.bdx != 0):
+    if (len(f.lbcode) == 5 and f.lbcode[-1] == 1 and f.lbcode.ix == 13):
         dim_coords_and_dims.append(
-            (DimCoord.from_regular(f.bzx, f.bdx, f.lbnpt,
-                                   long_name='site_number', units='1'),
+            (DimCoord(
+                f.x,
+                long_name='site_number',
+                units='1'),
              1))
 
+    if (len(f.lbcode) == 5 
+            and f.lbcode[-1] == 3 
+            and f.lbcode.ix == 13 
+            and hasattr(f, 'lower_z_domain') and all(f.lower_z_domain != -1.e+30)
+            or hasattr(f, 'lower_z_domain') and all(f.lower_z_domain != -1.e+30)):
+        if f.bzx == 0 and f.bdx == 0 and not hasattr(f, 'x'):
+            if hasattr(f, 'lower_z_domain'):
+                points = f.lower_z_domain
+            if hasattr(f, 'upper_z_domain'):
+                points = f.upper_z_domain
```

Comment:
Thinking about it, lower_ and upper_z_domain should behave similar to lower_ and upper_x_domain, i.e. they provide the bounds for the values. So for site number, they should always be consistent.